llama_stack/providers/remote/inference/fireworks/fireworks.py

```diff
@@ -236,14 +243,18 @@ async def _get_params(
         media_present = request_has_media(request)
 
         if isinstance(request, ChatCompletionRequest):
-            if media_present:
-                input_dict["messages"] = [
-                    await convert_message_to_openai_dict(m) for m in request.messages
+            input_dict["messages"] = [
+                await convert_message_to_openai_dict(m) for m in request.messages
+            ]
+            # print(input_dict["messages"])
+            if request.tool_choice == ToolChoice.required:
+                input_dict["tool_choice"] = "any"
+            
+            if request.tools:
+                input_dict["tools"] = [
+                    _convert_tooldef_to_openai_tool(t) for t in request.tools
                 ]
-            else:
-                input_dict["prompt"] = await chat_completion_request_to_prompt(
-                    request, self.get_llama_model(request.model), self.formatter
-                )
```

Comment:
ooc - why do we opt for using completions api and raw generations instead of using the apis directly?